sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2781,49 +2971,67 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    *   [[AS] select_statement];
    *
    *   create_table_clauses (order insensitive):
+   *     partition_clauses
    *     [OPTIONS table_property_list]
-   *     [PARTITIONED BY (col_name, transform(col_name), transform(constant, col_name), ...)]
+   *     [ROW FORMAT row_format]
+   *     [STORED AS file_format]
    *     [CLUSTERED BY (col_name, col_name, ...)
    *       [SORTED BY (col_name [ASC|DESC], ...)]
    *       INTO num_buckets BUCKETS
    *     ]
    *     [LOCATION path]
    *     [COMMENT table_comment]
    *     [TBLPROPERTIES (property_name=property_value, ...)]
+   *   partition_clauses:
+   *     [PARTITIONED BY (col_name, transform(col_name), transform(constant, col_name), ...)] |
+   *     [PARTITIONED BY (col2[:] data_type [COMMENT col_comment], ...)]
    * }}}
    */
   override def visitCreateTable(ctx: CreateTableContext): LogicalPlan = withOrigin(ctx) {
     val (table, temp, ifNotExists, external) = visitCreateTableHeader(ctx.createTableHeader)
-    if (external) {
-      operationNotAllowed("CREATE EXTERNAL TABLE ...", ctx)
-    }
-    val schema = Option(ctx.colTypeList()).map(createSchema)
+
+    val columns = Option(ctx.colTypeList()).map(visitColTypeList)
     val provider = Option(ctx.tableProvider).map(_.multipartIdentifier.getText)
-    val (partitioning, bucketSpec, properties, options, location, comment) =
+    val (partTransforms, partCols, bucketSpec, properties, options, location, comment, serdeInfo) =
       visitCreateTableClauses(ctx.createTableClauses())
 
+    if (provider.isDefined && serdeInfo.isDefined) {
+      operationNotAllowed(s"CREATE TABLE ... USING ... ${serdeInfo.get.describe}", ctx)
+    }
```

Comment:
We should add the temp table check here following https://github.com/apache/spark/pull/28026/files#diff-77a9aad2da3dc60210a2c4d2f3165d5f1d0acd54ca4811072a053225170ed748L396
```
    if (temp) {
      operationNotAllowed("CREATE TEMPORARY TABLE is not supported yet. " +
        "Please use CREATE TEMPORARY VIEW as an alternative.", ctx)
    }
```

The error message is better than https://github.com/apache/spark/pull/28026/files#diff-26fe6b511f4b8aedf5bdffae0d7c03aba69463c09fa4f881d77520125796e0faR3029

Since we are merging the syntax, we should pick the better error message from the hive syntax side.